src/Helpers.js

```diff
@@ -0,0 +1,44 @@
+
+// jsonTurnNumbersToStrings :: String -> String
+exports.jsonTurnNumbersToStrings = (str) => {
+    const s = String(str);
+
+    var arr = [];
+    var prev_in_number = false;
+    var in_number = false;
+    var in_string = false;
+
+    for (var i = 0, n = s.length; i < n; ++i) {
+        const c = s[i];
+        if (c == '"') {
+            if( i ==0 ||  (i>0 && s[i-1] != "\\" )){
```

Comment:
Looking back by 1 character is not enough, you need a flag:

```
> JSON.parse('{"a":"\\\\", "b": 123}')
{ a: '\\', b: 123 }
> jsonTurnNumbersToStrings('{"a":"\\\\", "b": 123}')
'{"a":"\\\\", "b": 123}'
```

Reply:
Would still be scary though!

I'm also not in favor of using that `json-bigint` library (it does not seem to be well tested either).

Our problem is:

> haskell serializes arbitrary-precision numbers as Integers.

I'm out of context w.r.t. what are we going to parse here. Still, what can be done on haskell side?

Reply:
> Would still be scary though!
> 
> I'm also not in favor of using that `json-bigint` library (it does not seem to be well tested either).
> 
> Our problem is:
> 
> > haskell serializes arbitrary-precision numbers as Integers.
> 
> I'm out of context w.r.t. what are we going to parse here. Still, what can be done on haskell side?
I'm also not really introduced to the haskell side.
cc @Benjmhart

Reply:
There are various fields we need to parse as `BigInt`s when we receive query responses from ogmios (e.g. the `coins` field of the `Value` type defined in ogmios). The encoders are in `Ogmios.Data.Json.Prelude`. For example, `Coin` uses `encodeInteger` (just an alias for the normal aeson `Integer` encoder). While we could change this, we would have to encode all `Coin`s as strings, but I believe this would entail more changes to the jsonwsp schema, etc...

Reply:
We want our ogmios fork to be as generally useful as possible, so I don't think we should change it there for what is really a Purescript-specific issue

Reply:
What if we re-parsed and re-encoded the jsons on haskell side just before sending them to frontend? I mean a thin layer in front of ogmios?

@ngua

Reply:
Apparently we should be able to traverse the jsons without the need to know the datatype schema that was used to generate them. The advantage is that we would be dealing with already-parsed jsons (as a tree structure), which is more reliable than a custom state machine-based tokenizer of string.